packages/shared_preferences/shared_preferences_tool/lib/src/ui/data_panel.dart

```diff
@@ -0,0 +1,326 @@
+                              _ => TextFormField(
+                                  autofocus: true,
+                                  initialValue: value.data.valueAsString,
+                                  inputFormatters: switch (value.data) {
+                                    SharedPreferencesDataInt() =>
+                                      <TextInputFormatter>[
+                                        FilteringTextInputFormatter.allow(
+                                          RegExp(r'^-?\d*'),
+                                        ),
+                                      ],
+                                    SharedPreferencesDataDouble() =>
+                                      <TextInputFormatter>[
+                                        FilteringTextInputFormatter.allow(
+                                          RegExp(r'^-?\d*\.?\d*'),
+                                        ),
+                                      ],
+                                    _ => <TextInputFormatter>[],
+                                  },
+                                  onChanged: _setCurrentValue,
+                                )
+                            },
+                          ] else ...<Widget>[
+                            Text('Value: ${value.data.valueAsString}'),
+                          ],
+                        ],
+                      ),
+                    ),
+                  ),
+                ),
+              ),
+            ],
+          ),
+      },
+    );
+  }
+}
+
+class _EditStringList extends StatefulWidget {
+  const _EditStringList({
+    required this.selectedKey,
+    required this.initialData,
+    required this.onChanged,
+  });
+
+  final String selectedKey;
+  final List<String> initialData;
+  final ValueChanged<String> onChanged;
+
+  @override
+  State<_EditStringList> createState() => _EditStringListState();
+}
+
+class _EditStringListState extends State<_EditStringList> {
+  late final List<(int key, String value)> _currentList;
+  int _keyCounter = 0;
+
+  void _addElementAt(int index) {
+    setState(() {
+      _currentList.insert(index, (_keyCounter++, ''));
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _currentList = <(int, String)>[
+      for (final String str in widget.initialData) (_keyCounter++, str),
+    ];
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      mainAxisSize: MainAxisSize.min,
+      children: <Widget>[
+        for (final (int index, (int keyValue, String str))
+            in _currentList.indexed) ...<Widget>[
+          _AddListElement(
+            onPressed: () => _addElementAt(index),
+          ),
+          Padding(
+            padding: const EdgeInsets.symmetric(vertical: densePadding),
+            child: Row(
+              children: <Widget>[
+                Expanded(
+                  child: TextFormField(
```

Comment:
same here. Give this height `defaultTextFieldHeight`

Reply:
same as above